amgut/lib/data_access/ag_data_access.py

```diff
@@ -389,25 +409,30 @@ def logParticipantSample(self, ag_login_id, barcode, sample_site,
                          WHERE ag_login_id = %s AND participant_name = %s"""
 
                 TRN.add(sql, (ag_login_id, participant_name))
-                survey_id = TRN.execute_fetchindex()
-                if not survey_id:
-                    raise ValueError("No survey ID for ag_login_id %s and "
+                survey_ids = [x[0] for x in TRN.execute_fetchindex()]
```

Comment:
I think in the case that there are no survey ids this will fail. I think the correct thing to do is:
```python
survey_ids = TRN.execute_fetchindex()
if not survey_ids:
    raise...
survey_ids = [x[0] for x in survey_ids]
```